### DISCUSSION TITLE
关于 relay 与 relax 的过渡

### DISCUSSION THREAD

**Discussion by xinetzone** [mlc-ai/mlc-zh#44] [Q&A]
请问当前 TVM 使用的是 relay，relay 与 relax 有什么区别？又该如何从 relay 过渡到 relax 呢？

**Comment by Hzfengsy** [comment]
relay和relax同为TVM图层IR。Relay设计较早，且作为函数式语言，存在一些设计上的缺陷（如dynamic shape），因此设计了relax。短期来看relay和relax会共存，给用户足够的时间进行升级

**Reply by xinetzone** [reply]
感谢解答。

relax 大概还需要多久合并到 TVM main?
